experimental/LieAlgebras/src/PathModel.jl

```diff
@@ -0,0 +1,617 @@
+##### Kashiwara Crystal #####
+
+abstract type AbstractCrystal <: AbstractAlgebra.Set end
+
+abstract type AbstractCrystalElem <: AbstractAlgebra.SetElem end
+
+# AbstractCrystalElem required methods
+
+function Base.eps(b::AbstractCrystalElem)
+  error("not implemented")
+end
+
+function phi(b::AbstractCrystalElem)
+  error("not implemented")
+end
+
+@doc raw"""
+    Apply the root operator $\tilde e_i$ to `b` in place and return the result.
+"""
+function ealpha!(b::AbstractCrystalElem, i::Int)
+  error("not implemented")
+end
+
+@doc raw"""
+    Apply the root operator $\tilde f_i$ to `b` in place and return the result.
+"""
+function falpha!(b::AbstractCrystalElem, i::Int)
+  error("not implemented")
+end
+
+@doc raw"""
+    Return the weight of `b`.
+"""
+function weight(b::AbstractCrystalElem)
+  error("not implemented")
+end
+
+# AbstractCrystal optional methods
+
+@doc raw"""
+    is_upper_normal(B::AbstractCrystal) -> bool
+
+Return `true` if `B` is upper normal, and `false` if `B` is not upper normal or it is unknown.
```

Comment:
Is this a standard term, that everyone working on these things would immediately understand? If not perhaps it could be briefly defined here?

Alternatively resp. in addition, a reference should be added to the bibliography that defines all these terms, and your `.md` file should point out this reference: "We follow the terminology and definitions in CITE".